js/web/lib/wasm/jsep/webgpu/gpu-data-manager.ts

```diff
@@ -223,11 +235,31 @@ class GpuDataManagerImpl implements GpuDataManager {
 
   refreshPendingBuffers(): void {
     for (const buffer of this.buffersForUploadingPending) {
+      // upload buffer is only useful in the session creation time. So we don't need to reuse them in session running.
       buffer.destroy();
     }
+    this.buffersForUploadingPending = [];
     for (const buffer of this.buffersPending) {
-      buffer.destroy();
+      // Put the pending buffer to freeBuffers list instead of really destroying it for buffer reusing.
+      this.freeBuffers.get(buffer.size)?.push(buffer);
```

Comment:
maybe change `?.` to `!.` as we think the map should always contain value for key `buffer.size`

Reply:
Done.